### PR TITLE
Fix Sphinx>4 warnings with units and coordinates

### LINKED ISSUE #11725: DOC: Unpin max version of Sphinx when Sphinx 4.0.2 is out (and Jinja2)
**Update:** If you want to build docs in Python 3.10 or newer, see the pins in  #13587 .

Revert #11724 when sphinx-doc/sphinx#9210 is resolved. @saimn tracked it down to sphinx-doc/sphinx#8935 .

https://github.com/astropy/astropy/blob/642d8aea136d274a82928a8735cd25d6905116ed/setup.cfg#L84-L85

Also revert #12988 .

https://github.com/astropy/astropy/blob/f6569ceaeb81afef80d5a6a0a579698f1c27c6cd/setup.cfg#L103

### PR DESCRIPTION
**Pull request by saimn**
Another attempt to fix #11725.

### Description

This pull request is to address ...

Fixes #<Issue Number>

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by pllim** [comment]
Please don't tell me this is April Fool PR. My heart can't take it. It looks so elegant!

**Comment by pllim** [comment]
Don't worry about mpl or numpy dev failures. They are not related.

**Comment by pllim** [comment]
RTD is green! 🎉

**Review by pllim — APPROVED** [review]
The diff LGTM but coordinates maintainers (particularly @adrn ?) should see if the rendered API doc is still ok.

**Comment by pllim** [comment]
cc @maxnoe

**Comment by maxnoe** [comment]
But that just removes the corresponding documentation. This is not what we want, right? We want a way to avoid these warnings without loosing the documentation pages.

**Comment by saimn** [comment]
Yes, so the idea was first to see if this fixes the warnings. Those functions are already documented at the top level, and are duplicated in a specific section.
For `coordinates` documenting the built-in frames in a specific section makes a lot of sense but I don't think there is way currently to remove this module from the top-level documentation (to avoid the duplicate warning). I think it's possible to skip a class/function but not a whole module ? Maybe something we can improve in sphinx-automodapi.
For `units` it's less clear to me, there are other modules that are duplicated but do not raise the warning ...

**Comment by pllim** [comment]
IMHO the benefit of getting rid of two maxversion pins outweighs the cost of losing some duplicate doc in coordinates. I have no idea when sphinx-automodapi will get fixed and this has been plaguing us long enough.

Can we go ahead with this and then open up follow-up issue (or comment on existing ones)?

File: docs/units/index.rst [hunk 1]
```diff
@@ -254,11 +254,6 @@ Reference/API
 
 .. automodapi:: astropy.units.equivalencies
 
-.. automodapi:: astropy.units.function
```
Comment (mhvk) [inline docs/units/index.rst:257]: why are these removed? They are distinct from `astropy.units.function.units` and, e.g., `STmag` does not seem to be linked anywhere else on the present doc pages.

Comment (saimn) [reply docs/units/index.rst:257]: I think I had to remove this one as well to fix the warnings.

Comment (mhvk) [reply docs/units/index.rst:257]: And how about `function.logarithmic`? Is removing that truly needed? If present, what is the error?

File: docs/coordinates/index.rst [hunk 2]
```diff
@@ -547,11 +547,6 @@ coordinate systems implemented here include:
 Built-in Frame Classes
 ======================
 
-.. automodapi:: astropy.coordinates.builtin_frames
```
Comment (mhvk) [inline docs/coordinates/index.rst:550]: I guess a similar question here - certainly, with this removed, the section title should be removed too -- but this particular line is a very big deal in the actual docs!! https://docs.astropy.org/en/latest/coordinates/index.html#built-in-frame-classes

**Comment by pllim** [comment]
As discussed at infrastructure tag-up today (2022-04-07) between @saimn , @eteq , @bsipocz , @tomdonaldson , and myself, here is the plan:

1. @eteq will look into a proper upstream fix for astropy/sphinx-automodapi#130 (some useful discussion in that issue including https://github.com/maxnoe/sphinx-automodapi-duplicated-warning to reproduce this outside of astropy) or possibility of astropy/sphinx-automodapi#147
2. If proper fix not implemented by v5.2 feature freeze time (end of 2022), we clean up and merge PR at the cost of some documentation loss.

Stretch goal:

* astropy/sphinx-automodapi#146 if astropy/sphinx-automodapi#147 does not happen

**Comment by saimn** [comment]
See https://github.com/sphinx-doc/sphinx/issues/10348#issuecomment-1100513041 for a great description of the problem and the different options to solve this. Closing, we should continue the discussion in https://github.com/astropy/sphinx-automodapi/issues/130.
